packages/vite/src/node/plugins/worker.ts

```diff
@@ -204,16 +217,12 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
       })
     },
 
+    shouldTransformCachedModule({ id }) {
+      return config.build.lib && load(id)
```

Comment:
```suggestion
      return load(id)
```
It happens not only with lib mode.
https://stackblitz.com/edit/vitejs-vite-eciitf?file=package.json (run both `build` and `preview`)